types/src/consensus/attestation_metadata.rs

```diff
@@ -9,3 +9,27 @@ pub struct AttestationMetadata {
     pub global_sequence: u64,
     pub contract_sequences: Vec<(Address, u64)>,
 }
+impl Hashable for AttestationMetadata {
+    fn hash_custom(&self) -> Hash {
+        use alloy_primitives::Keccak256;
+
+        let mut hasher = Keccak256::default();
+        hasher.update(self.timestamp.hash_custom().as_slice());
+        hasher.update(self.global_sequence.to_be_bytes());
+
+        //sort to ensure deterministic, order-independent hashing
+        let mut sorted = self.contract_sequences.clone();
+        sorted.sort_by(|a, b| match a.0.cmp(&b.0) {
+            std::cmp::Ordering::Equal => a.1.cmp(&b.1),
+            other => other,
+        });
```

Comment:
Is it even valid to have the same address twice in `contract_sequence`? Perhaps it should be a `BTreeMap`, which already gives: deterministic order and uniqueness, wdyt?

Reply:
No it is invalid to have duplicate contract. BTreeMap is indeed a good option.
Although, I am skeptical. I believe that maps and sets are avoided by convention.
Network messages should be a simple as possible.

What if validators return sorted vec instead?

Reply:
Just checked code of validator: the contract_sequences `Vec` is guaranteed to come out in deterministic, sorted order.
Therefore I suggest to keep it as it is (and remove the sorting in hashing), wdyt?